backend/src/transaction_event/protocol_level_tokens.rs

```diff
@@ -625,39 +639,217 @@ impl PreparedTokenUpdate {
+                                token_idx,
+                                $5,
+                                (SELECT decimal FROM plt_tokens WHERE token_id = $3)
+                            FROM account_checks
+                            ON CONFLICT (account_index, token_index) DO UPDATE
+                            SET amount = plt_accounts.amount + $5
+                            RETURNING amount as to_balance
+                        )
+                        SELECT 
+                            token_idx, 
+                            from_existed, 
+                            to_existed,
+                            COALESCE((SELECT from_balance FROM balance_updates), 0) as \
+                         from_final_balance
+                        FROM account_checks
+                        ",
+                        from,
+                        to,
+                        self.token_id,
+                        -&self.plt_amount_change, // Negative amount for from account
+                        self.plt_amount_change    // Positive amount for to account
+                    )
+                    .fetch_one(tx.as_mut())
+                    .await?;
+
+                    let from_existed = result.from_existed.unwrap_or(false);
+                    let to_existed = result.to_existed.unwrap_or(false);
+                    let from_final_balance =
+                        result.from_final_balance.unwrap_or_else(|| BigDecimal::from(0));
+
+                    // Calculate unique account changes: +1 for new accounts, -1 for zero balances
+                    let mut new_accounts = 0i64;
+                    if !from_existed {
+                        new_accounts += 1;
+                    }
+                    if !to_existed {
+                        new_accounts += 1;
+                    }
+                    if from_existed && from_final_balance == BigDecimal::from(0) {
+                        new_accounts -= 1;
+                    }
+
+                    let token_idx = result
+                        .token_idx
+                        .ok_or_else(|| anyhow::anyhow!("Token not found: {}", self.token_id))?;
+                    // normalize amount for cross-token consistency
+                    let normalized_amount = self.plt_amount_change.clone()
+                        / BigDecimal::from(10u64.pow(self.amount_decimals as u32));
+
+                    // METRICS UPDATE 1: Global PLT transfer aggregation
+                    //
+                    // Updates metrics_plt table with  transfer data:
+                    // - cumulative_transfer_amount: Running sum of normalized transfer volumes
+                    // - unique_account_count: Count of distinct accounts across all PLT tokens
+                    //
+                    // Query execution:
+                    // - CTE fetches previous metric values from latest row
+                    // - INSERT with ON CONFLICT (event_timestamp) for idempotent updates
```

Comment:
I think a better explanation is that you insert max one row per block in the `metrics_plt` table.
Meaning, a new row in the `metrics_plt` table is inserted for the first PLT event in the block. For any further PLT event/update in the same block, the row is updated instead.

What do you mean with the `idempotent` exactly as the row will be changed/updated (you probably want to say that no new row will be generated in the `metrics_plt` table if several PLT events exist per block/event_timestamp/block_timestamp.